src/libs/DistanceRequestUtils.ts

```diff
@@ -256,11 +263,22 @@ function getCustomUnitRateID(reportID: string) {
     const report = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] ?? null;
     const parentReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${report?.parentReportID}`] ?? null;
     const policy = PolicyUtils.getPolicy(report?.policyID ?? parentReport?.policyID ?? '');
-
     let customUnitRateID: string = CONST.CUSTOM_UNITS.FAKE_P2P_ID;
 
+    if (isEmptyObject(policy) || !policy?.customUnits) {
+        return customUnitRateID;
+    }
```

Comment:
I don't really like this condition, because we could have a policy without custom units which is now oddly returning the fake id. I doubt that would really happen, but overall this condition is confusing and seems unnecessary.

I think we can handle the rest fine even if policy or customUnits is undefined. Let's move getting the lastSelectedDistanceRateID into the condition where we know it's a policy expense chat.